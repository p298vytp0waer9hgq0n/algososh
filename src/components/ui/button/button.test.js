import renderer from "react-test-renderer";

import { Button } from "./button";

describe(Button, () => {
    it('с текстом', () => {
        const tree = renderer.create(<Button>with text</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('без текста', () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('заблокирована', () => {
        const tree = renderer.create(<Button disabled={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('с индикатором загрузки', () => {
        const tree = renderer.create(<Button isLoader={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});