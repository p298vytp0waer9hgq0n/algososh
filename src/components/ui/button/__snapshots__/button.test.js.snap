// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button заблокирована 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button с индикатором загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  with text
  <p
    className="text undefined"
  />
</button>
`;
